dev/scripts/build-docker.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+#
+# The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+# (the "License"). You may not use this work except in compliance with the License, which is
+# available at www.apache.org/licenses/LICENSE-2.0
+#
+# This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied, as more fully set forth in the License.
+#
+# See the NOTICE file distributed with this work for information regarding copyright ownership.
+#
+
+#
+# This script generates a Docker image for the repository from its current state.
+#
+
+set -e
+
+this=$(cd "$( dirname "$0" )"; pwd)
+cd ${this}
```

Comment:
applies throughout this PR ... basically, you want to be able to support paths that have space in them ... if you don't quote paths, your script will be unhappy (or worse, silently do the wrong thing)